src/compas_fab/robots/robot.py

```diff
@@ -46,11 +46,21 @@ class Robot(object):
     """
 
     def __init__(self, model, artist=None, semantics=None, client=None):
+        self._scale_factor = 1.
         self.model = model
-        self.artist = artist
+        self._artist = None
```

Comment:
I don't think you need to explicitly set it to `None` if the next line uses the property setter that assigns it anyway.